utils/compare-output/check.py

```diff
@@ -0,0 +1,178 @@
+#!/usr/bin/env python
+#
+# Copyright 2015 Delft University of Technology
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from __future__ import print_function
+from itertools import combinations
+import argparse
+import math
+import sys
+
+# Threshold use when comparing two floating-point values
+COMPARISON_THRESHOLD = 0.01
+
+
+def parse_file(filename, value_parser, verbose):
+    results = {}
+    errors = 0
+
+    with open(filename) as handle:
+        for index, line in enumerate(handle):
+            parts = line.strip().split()
+
+            try:
+                if parts and not parts[0].startswith('#'):
+                    if len(parts) != 2:
+                        raise ValueError('invalid number of fields')
+
+                    try:
+                        key = int(parts[0])
+                    except:
+                        raise ValueError('failed to parse vertex id '
+                                '("{}" is not a number)'
+                                .format(parts[0]))
+
+                    try:
+                        value = value_parser(parts[1])
+                    except:
+                        raise ValueError('failed to parse vertex value '
+                                '("{}" is not a value)'
+                                .format(parts[1]))
+
+                    if key in results:
+                        raise ValueError('duplicate entry for vertex {}'
+                                .format(key))
+
+                    results[key] = value
+
+            except ValueError as e:
+                if verbose:
+                    print('error in {} on line {}: {}'.format(filename,
+                        index + 1, e), file=sys.stderr)
+
+                errors += 1
+
+    return results, errors
+
+
+def check_exact():
+    return lambda x, y: x == y
+
+
+def check_threshold():
+    def comparison(x, y):
+        global COMPARISON_THRESHOLD
+
+        # Check exact match first to handle +/-inf
+        return x == y or abs(x-y) < COMPARISON_THRESHOLD
```

Comment:
This comparison does not match the definition from the benchmark specification, which checks if values are within a percentage (0.01%) of each other, rather than checking for absolute differences.